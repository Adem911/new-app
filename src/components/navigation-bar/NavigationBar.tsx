import React from "react";
import { Link } from "react-router-dom";
import { Menu, Icon } from "antd";
import "antd/dist/antd.css";

class NavigationBar extends React.Component {
  state = {
    current: "home"
  };

  handleClick = (e: any) => {
    this.setState({
      current: e.key
    });
  };

  render() {
    return (
      <Menu
        onClick={this.handleClick}
        selectedKeys={[this.state.current]}
        mode="horizontal"
      >
        <Menu.Item key="home">
          <Link to="/">
            <Icon type="mail" />
            Home
          </Link>
        </Menu.Item>

        <Menu.Item key="page1">
          <Link to="/page1">
            <Icon type="appstore" />
            Page One
          </Link>
        </Menu.Item>

        <Menu.Item key="page2">
          <Link to="/page2">
            <Icon type="appstore" />
            Page Two
          </Link>
        </Menu.Item>

        <Menu.Item key="page3">
          <Link to="/page3">
            <Icon type="appstore" />
            Page Two
          </Link>
        </Menu.Item>
      </Menu>
    );
  }
}

export default NavigationBar;
