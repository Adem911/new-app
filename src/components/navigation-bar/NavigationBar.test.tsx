import React from 'react';
import { shallow } from 'enzyme';
import NavigationBar from './NavigationBar';
import { Menu } from 'antd';

describe('<NavigationBar />', () => {
    it('renders correctly a navbar on top', () => {
        const wrapper = shallow(<NavigationBar />);
        expect(wrapper).toMatchSnapshot();

    })

    it('renders a navbar on top', () => {
        const wrapper = shallow(<NavigationBar />);
        expect(wrapper.find(Menu).length).toEqual(1);
        expect(wrapper.find(Menu.Item).length).toEqual(4);

    })
})

