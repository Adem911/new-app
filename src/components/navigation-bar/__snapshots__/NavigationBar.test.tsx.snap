// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> renders correctly a navbar on top 1`] = `
<Menu
  mode="horizontal"
  onClick={[Function]}
  selectedKeys={
    Array [
      "home",
    ]
  }
>
  <MenuItem
    key="home"
  >
    <Link
      to="/"
    >
      <Icon
        type="mail"
      />
      Home
    </Link>
  </MenuItem>
  <MenuItem
    key="page1"
  >
    <Link
      to="/page1"
    >
      <Icon
        type="appstore"
      />
      Page One
    </Link>
  </MenuItem>
  <MenuItem
    key="page2"
  >
    <Link
      to="/page2"
    >
      <Icon
        type="appstore"
      />
      Page Two
    </Link>
  </MenuItem>
  <MenuItem
    key="page3"
  >
    <Link
      to="/page3"
    >
      <Icon
        type="appstore"
      />
      Page Three
    </Link>
  </MenuItem>
</Menu>
`;
